account_bank_statement_import_txt_xlsx/models/account_bank_statement_import_sheet_parser.py

```diff
@@ -204,15 +223,23 @@ def _parse_lines(self, mapping, data_file, currency_code):
                     timestamp,
                     mapping.timestamp_format
                 )
-
-            amount = self._parse_decimal(amount, mapping)
-            if balance is not None:
-                balance = self._parse_decimal(balance, mapping)
-
+            if amount_column:
+                amount = self._parse_decimal(
+                    values[amount_column], mapping)
             if debit_credit is not None:
                 amount = amount.copy_abs()
                 if debit_credit == mapping.debit_value:
                     amount = -amount
+            if debit_column and credit_column:
+                debit_amount = self._parse_decimal(
+                    values[debit_column], mapping)
+                debit_amount = debit_amount.copy_abs()
+                credit_amount = self._parse_decimal(
+                    values[credit_column], mapping)
+                amount = -debit_amount or credit_amount
```

Comment:
I'd suggest: `amount = credit_amount - debit_amount` as empty values are now clean zeros, and a code reader will spend time wondering (as I just did) if the `debit_amount == 0` case (which lead to `amount = credit_amount` in the above code) is well handled.